allennlp/tests/predictors/bidaf_test.py

```diff
@@ -96,3 +96,39 @@ def test_model_internals(self):
         # hooks should be gone
         for module in predictor._model.modules():
             assert not module._forward_hooks
+
+    def test_predictions_to_labeled_instances(self):
+        inputs = {
+                "question": "What kind of test succeeded on its first attempt?",
+                "passage": "One time I was writing a unit test, and it succeeded on the first attempt."
+        }
+
+        archive = load_archive(self.FIXTURES_ROOT / 'bidaf' / 'serialization' / 'model.tar.gz')
+        predictor = Predictor.from_archive(archive, 'machine-comprehension')
+
+        instance = predictor._json_to_instance(inputs)
+        outputs = predictor._model.forward_on_instance(instance)
+        new_instances = predictor.predictions_to_labeled_instances(instance, outputs)
+        assert 'span_start' in new_instances[0].fields
+        assert 'span_end' in new_instances[0].fields
+        assert new_instances[0].fields['span_start'] is not None
+        assert new_instances[0].fields['span_end'] is not None
+
+    def test_get_gradients(self):
```

Comment:
Because this is testing a function in the base class, I'd rather see this test in the [base class test file](https://github.com/allenai/allennlp/blob/master/allennlp/tests/predictors/predictor_test.py).  And then you also don't need to repeat it for every predictor.

Reply:
each predictor is slightly different though, some have grad_input_2, etc.

Reply:
But they are all using the same implementation.  If you test it on the base class, with one concrete implementation that has more than one input, that is sufficient.